common/transport/amqp/devdoc/amqp_cbs_requirements.md

```diff
@@ -0,0 +1,100 @@
+# ReceiverLink Requirements
+
+## Overview
+
+The `ReceiverLink` class is internal to the SDK and shall be considered private. It shall not be used directly by clients of the SDK, who should instead rely on the higher-level constructs (`Client` classes provided by `azure-iot-device` and `azure-iothub`).
+
+The `ReceiverLink` class implements a state machine that manages the underlying `amqp10` link object used to receive messages from IoT Hub. It can be attached and detached manually, and will try to attach automatically if not already attached when setting up a message listener or trying to settle an existing message.
+
+## Example usage
+
+```typescript
+import * as amqp10 from 'amqp10';
+import { ClaimsBasedSecurityAgent } from './ampq_cbs';
+
+const amqp10Client = new amqp10.AmqpClient(null);
+// the amqp10 client shall be connected before using it with the CBS agent.
+const cbs = new ClaimsBasedSecurityAgent(amqp10Client);
+cbs.attach((err) => {
+  if (err) {
+    console.error(err.toString());
+  } else {
+    cbs.putToken('<IoT Hub hostname', 'shared access signature', (err) => {
+      if (err) {
+        console.log(err.toString());
+      } else {
+        // now authenticated using CBS
+      }
+    });
+  }
+});
+```
+
+## Public Interface
+
+### constructor(amqpClient)
+
+**SRS_NODE_AMQP_CBS_16_001: [** The `constructor` shall instanciate a `SenderLink` object for the `$cbs` endpoint using a custom policy `{encoder: function(body) { return body;}}` which forces the amqp layer to send the token as an amqp value in the body. **]**
+
+**SRS_NODE_AMQP_CBS_16_002: [** The `constructor` shall instanciate a `ReceiverLink` object for the `$cbs` endpoint. **]**
+
+**SRS_NODE_AMQP_CBS_16_017: [** The `ClaimsBasedSecurityAgent` class shall inherit from the native `EventEmitter` class. **]**
+
+### attach(callback)
+
+**SRS_NODE_AMQP_CBS_16_003: [** `attach` shall attach the sender link. **]**
+
+**SRS_NODE_AMQP_CBS_16_004: [** `attach` shall attach the receiver link. **]**
+
+**SRS_NODE_AMQP_CBS_16_005: [** The `attach` method shall set up a listener for responses to put tokens on the `message` event of the receiver link. **]**
+
+**SRS_NODE_AMQP_CBS_16_006: [** If given as an argument, the `attach` method shall call `callback` with a standard `Error` object if any link fails to attach. **]**
+
+**SRS_NODE_AMQP_CBS_16_007: [** If given as an argument, the `attach` method shall call `callback` with a `null` error object if successful. **]**
+
+### detach()
+
+**SRS_NODE_AMQP_CBS_16_008: [** `detach` shall detach both sender and receiver links and return the state machine to the `detached` state. **]**
```

Comment:
>e state machine to the `detached` state. [](start = 98, length = 40)

You never mention states anywhere else.  Did you want to say that the constructor starts the fsm in the detached state?

You also never discuss leaving the detached state for any other state.

Is it worth putting in the transition diagram for the FSM?